rs/protobuf/def/registry/node/v1/node.proto

```diff
@@ -46,6 +63,8 @@ message NodeRecord {
   // If a Node is to be converted into the ApiBoundaryNode, the domain field should be set.
   optional string domain = 19;
 
+  optional NodeType node_type = 20;
```

Comment:
Does it need to be optional ? Can't we use unspecified when missing ?

Reply:
https://protobuf.dev/programming-guides/dos-donts/#:~:text=the%20old%20one.-,Don't%20Add%20a%20Required%20Field,were%20removed%20from%20proto3%20completely.

Reply:
The recommendation makes sense. However I think it is different from what I am proposing. We are not making the node_type type required. We just treat it similar to a string.

For example if you have a string field that is not optional the default value is empty string. Similarly here, if this field is non optional than the default value will be unspecified.

Reply:
Or let me put it differently to my understanding None and Unspecified are semantically the same if I understand correctly.

Reply:
The unspecified is to follow the style guide that we seem to have adopted.  But we also independently decided to make fields optional to make it easier to migrate.  I think this may have to do with the generated rust code sometimes ending up in candid endpoints, where the rules for how things are encoded/decoded and what to use as defaults is different than protobuf.  

So - we just decided to make all the protobuf fields optional, as it makes it easier to migrate regardless.

Our future plan is to not expose generated protobuf as parts of candid interfaces so that we have proper API-level encapsulation of the internals of canisters, but in places where we have protobuf, we decided to just make things optional so we don't have to think harder about it.